_posts/2020-08-10-deepfake/deepfake.Rmd

```diff
@@ -0,0 +1,231 @@
+---
+title: "Deepfake detection challenge from R"
+description: |
+  A couple of months ago AWS, Facebook, Microsoft, and other contributors brought a challenge where users have to build a model to differentiate very realistic AI-generated fake videos. Despite the complexity and uniqueness of the computer vision task, we will show how the R community can also work with video dataset and bring their own solution.
+author:
+  - name: Turgut Abdullayev 
+    url: https://github.com/henry090
+    affiliation: QSS Analytics
+    affiliation_url: http://www.qss.az/
+date: 08-11-2020
+categories:
+  - Image Recognition & Image
+creative_commons: CC BY
+repository_url: https://github.com/henry090/Deepfake-from-R
+output: 
+  distill::distill_article:
+    self_contained: false
+    toc: true
+    toc_depth: 2
+preview: files/frame_2.png
+---
+
+
+
+<style type="text/css">
+.colab-root {
+    display: inline-block;
+    background: rgba(255, 255, 255, 0.75);
+    padding: 4px 8px;
+    border-radius: 4px;
+    font-size: 11px!important;
+    text-decoration: none;
+    color: #aaa;
+    border: none;
+    font-weight: 300;
+    border: solid 1px rgba(0, 0, 0, 0.08);
+    border-bottom-color: rgba(0, 0, 0, 0.15);
+    text-transform: uppercase;
+    line-height: 16px;
+}
+span.colab-span {
+    background-image: url(https://distill.pub/2020/growing-ca/images/colab.svg);
+    background-repeat: no-repeat;
+    background-size: 20px;
+    background-position-y: 2px;
+    display: inline-block;
+    padding-left: 24px;
+    border-radius: 4px;
+    text-decoration: none;
+}
+</style>
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+## Introduction
+
+Working with video datasets, particularly with detection of AI-based fake objects is very challenging due to proper frame selection and face detection. However, the solution for this case can be found from the R side as well. For this purpose, one can benefit the opportunities of OpenCV, magick, and Keras libraries from R.
+
+Before going into detailed explanation and code exploration, readers should know that there is no need to copy-paste code chunks. Because at the end of the post one can find a link to Google Colab with GPU acceleration. This Kernel allows everyone to run and reproduce the same results.
+
+
+## Data exploration
+
+[The dataset](https://www.kaggle.com/c/deepfake-detection-challenge/overview/description) that we are going to analyze is provided by AWS, Facebook, Microsoft, the Partnership on AI’s Media Integrity Steering Committee, and academics.
+
+It contains both real and AI-generated fake videos. The total size is over 470 GB. However, the sample 4 GB dataset is separately available.
+
+## Frame extraction
+
+The videos in the folders are in the format of _mp4_ and have various lengths. The caption of the right frames and definition of total frames for each second is out of the scope of this post. We usually took 1-3 fps for every video.
```

Comment:
```suggestion
The videos in the folders are in _mp4_ format and have various lengths. The caption of the right frames and definition of total frames for each second is out of the scope of this post. We usually took 1-3 fps for every video.
```

Reply:
what does this mean: "The caption of the right frames and definition of total frames for each second"?

If you say it's out of scope, can you indicate a place where this is discussed? And say which part of the code handles this?